### DISCUSSION TITLE
Flutter iOS build Failed - No profiles for were found: Xcode couldn't find any iOS App Development provisioning profiles matching

### DISCUSSION THREAD

**Discussion by MohammadDiab** [codemagic-ci-cd/codemagic-docs#2128] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1- Build triggers
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/8952135/878123af-deaf-4fb5-8fce-cf3954941256)
2- Environment variables
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/8952135/57705c59-6793-4f8e-b1bb-7c6015835231)
3- Build
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/8952135/9a290e87-986a-4fbf-a1df-6c6dd251bd81)
4- Distribution
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/8952135/325f86aa-98be-4676-924d-6b5fcc469fdf)
5- App Store Connect
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/8952135/d8699f05-bcb1-402f-9aa4-97d161b3dd2d)

### Expected results

Build should be succeed and ready to publish to store 

### Actual results

Build failed 
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/8952135/5787bc0e-0ecd-42e1-bf70-537d53a376d0)
Error (Xcode): No profiles for 'xxx.xxx.xxx' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'xxx.xxx.xxx'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
/Users/builder/clone/ios/Runner.xcodeproj

Error (Xcode): No profiles for 'xxx.xxx.xxx.onesignal' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'xxx.xxx.xxx.onesignal'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
/Users/builder/clone/ios/Runner.xcodeproj

### Build id (optional)

652f91f756c76347589e53dd

**Comment by MohammadDiab** [comment]
Any Update?

**Reply by rafeysup93** [reply]
Hi @MohammadDiab,

In this issue, there are two errors. The first one is related to the provisioning profile of your application, and the second error is related to the app extensions, which is OneSignal in this case. 

1- For the first error related to your app:

Please check your bundle identifiers. Your project bundle ID must match the one in your App Store Connect app bundle ID. Also, review your Xcode locally to see if you have the right Bundle ID there.

<img width="756" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/127734704/fb10da24-0a97-483f-be06-79d80074a2de">

2- For the second error:

If your app contains app extensions, an additional provisioning profile is required for each extension. Codemagic will use the bundle identifier to find the relevant provisioning profiles. If your bundle identifier is com.example.app, the matching profiles are the ones with com.example.app and com.example.app.* as bundle identifier.

**Reply by MohammadDiab** [reply]
@rafeysup93 First of all I used codemagic to build and publish my app since I don't have Xcode and mac machine, second thing I added all required profiles to codemagic can you please check the Build Id Again

**Comment by MonsterDKing** [comment]
@rafeysup93 

hello i have the same issue : 


Archiving xxxxxx...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Automatically signing iOS for device deployment using specified development team in Xcode project: xxxxx
Running pod install...                                             24.4s
Running Xcode build...                                          
Xcode archive done.                                          3.5s
Failed to build iOS app
Error (Xcode): No profiles for 'xxxxx' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'xxxxxxx'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
/Users/builder/clone/ios/Runner.xcodeproj


It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.

ID BUILD : 6536d2d070b1e37fbe3a54df

I have been working for months without any problem. Just today I got this error, any suggestions, I already checked the xcode and if it has the signed version of iOS since it acquires it through the Apple Connect API and does it in adhoc to distribute for firebase distribution

**Comment by MohammadDiab** [comment]
@rafeysup93 Any update? it seems the issue is common issue

**Reply by rafeysup93** [reply]
Please verify that the Bundle Identifier in your project matches the one associated with your App Store Connect account. When selecting the profile in the distribution section, Codemagic displays all the available bundle IDs associated with your chosen App Store Connect API key. If you can't find your app's bundle identifier listed, create one in [Apple Developer Portal](https://developer.apple.com/account/resources/identifiers/add/bundleId).

<img width="762" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/127734704/bdc3c6c5-501d-4eb6-a4b6-819f750a4c7d">

Also, please be aware that Bundle Identifiers are case-sensitive.

To ensure your Bundle ID is correctly configured, please verify that it matches the one specified in the ios >> Runner.xcodeproj >> project.pbxproj file. You can locate this by searching for the key PRODUCT_BUNDLE_IDENTIFIER.

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.
